src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3WebtransportSession.cs

```diff
@@ -0,0 +1,196 @@
+            {
+                await DisposeAsync().ConfigureAwait(false);
+
+            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Current);
+
+            _ = _connectStream.ReadsClosed.ContinueWith(async t =>
+            {
+                await DisposeAsync().ConfigureAwait(false);
+
+            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Current);
+
+        }
+
+        /// <summary>
+        /// Creates a webtransport session by creating a webtransport connect request, sending it to <see cref="Uri">uri</see>.
+        /// </summary>
+        public static async ValueTask<Http3WebtransportSession?> ConnectAsync(Uri uri, HttpMessageInvoker invoker, CancellationToken cancellationToken)
+        {
+            Http3WebtransportSession? webtransportSession;
+            HttpRequestMessage request;
+            request = new HttpRequestMessage(HttpMethod.Connect, uri) { Version = HttpVersion.Version30, VersionPolicy = HttpVersionPolicy.RequestVersionExact };
+            request.Headers.Protocol = WebTransportProtocolValue;
+            Task<HttpResponseMessage> sendTask = invoker is HttpClient client ? client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken) : invoker.SendAsync(request, cancellationToken);
+            HttpResponseMessage response = await sendTask.ConfigureAwait(false);
+            WebtransportHttpContent connectedWebtransSessionContent = (WebtransportHttpContent)response.Content;
+            webtransportSession = connectedWebtransSessionContent.webtransportSession;
+            if (!response.IsSuccessStatusCode || !response.Headers.Contains(Http3WebtransportSession.VersionHeaderPrefix))
+            {
+                await webtransportSession.AbortIncomingSessionWebtransportStreamsAsync((long)Http3ErrorCode.WebtransportBufferedStreamRejected, cancellationToken).ConfigureAwait(false);
+                await webtransportSession.DisposeAsync().ConfigureAwait(false);
+                throw new HttpRequestException(SR.net_webtransport_server_rejected);
+            }
+
+            return webtransportSession;
+        }
+
+        /// <summary>
+        /// Takes the next incoming <see cref="QuicStream">quic stream from the server</see>.
+        /// </summary>
+        public async ValueTask<QuicStream?> GetIncomingWebtransportStreamFromServerAsync(CancellationToken cancellationToken = default)
+        {
+            if (_disposed == 1)
+                throw new ObjectDisposedException(nameof(Http3WebtransportSession));
+            try
+            {
+                QuicStream quicStream = await _incomingStreamsQueue.Reader.ReadAsync(cancellationToken).ConfigureAwait(false);
+                return quicStream;
+
+            }
+            catch (ChannelClosedException)
+            {
+                return null;
+            }
+            catch (OperationCanceledException)
```

Comment:
You shouldn't just swallow exceptions like this. You should let the user know what happened and why there are no more streams available. Especially OCE from cancellation token, it should bubble up to the user, as it is usually user-triggered and would have a separate handling.

As for other exceptions, they might just left as they are (not caught at all) or they can be wrapped in another exception for usability (like HttpRequestException wraps a lot of other exceptions) -- with the original exception usually being present as InnerException. It is very rare when you actually need to swallow exception. 

You can get some inspiration on what to do in e.g. QuicConnection
https://github.com/dotnet/runtime/blob/c39f2b69e1e8f643b1009088b49dd572506e3f42/src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs#L396-L404

Reply:
Fixed it, made the exceptions be thrown